### PR TITLE
'Registration form, user creation, authorization form, user authorization, navigation'

### PR DESCRIPTION
**Pull request by sasha53545**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,11 +3,15 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@material-ui/core": "^4.9.5",
+    "@material-ui/icons": "^4.9.1",
```
Comment (obabichev) [inline package.json]: This dependencies might be deleted. At least, I haven't found any usages in the code

File: src/App.module.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.header {
+
+}
+
+.main {
+
+}
```
Comment (obabichev) [inline src/App.module.css]: This file also might be deleted

File: src/components/SignIn/SignIn.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        fetch('/auth/login', {
+            method: 'POST',
+            headers: {
+                'content-type': 'application/json'
+            },
+            body: JSON.stringify(this.state.form)
+        })
+            .then(response => response.json())
+            .then(response => {
+                if(!response.status){
```
Comment (obabichev) [inline src/components/SignIn/SignIn.jsx]: Here is already parsed json from response body. You can not have status here (except there is a field `status` in the json, but I'm not sure that's what you want). You can check status in the previous `.then`)

File: src/components/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    onChange = (event) => {  //Как стрелочная функция заменяет bind();
+        this.setState({
+            form: {
+                ...this.state.form,     //Еще раз про спред оператор
+                [event.target.name]: event.target.value     //Почему ставим квадратные скобки
+            }
+        });
+        console.log('this.state.form: ', this.state.form);
```
Comment (obabichev) [inline src/components/SignUp/SignUp.jsx]: Please, avoid testing console logs in PR

File: src/components/SignUp/SignUp.jsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    };
+
+    onSubmit = (event) => {
+        if(event) {
+            event.preventDefault(); //Что еще интересно можно узнать про методы событий
```
Comment (obabichev) [inline src/components/SignUp/SignUp.jsx]: https://ru.reactjs.org/docs/events.html

```
boolean bubbles
boolean cancelable
DOMEventTarget currentTarget
boolean defaultPrevented
number eventPhase
boolean isTrusted
DOMEvent nativeEvent
void preventDefault()
boolean isDefaultPrevented()
void stopPropagation()
boolean isPropagationStopped()
DOMEventTarget target
number timeStamp
string type
```

Looks like there are not too many methods there

**Comment by obabichev** [comment]
Please, reopen PR into the branch created for you.

File: src/components/SignIn/SignIn.jsx (hunk 3 above)
Comment (sasha53545) [reply src/components/SignIn/SignIn.jsx]: fetch('/auth/register', {
          method: 'POST',
          headers: {
              'content-type': 'application/json'
          },
          body: JSON.stringify(this.state.form)
      })
          .then(response => {
              this.setState({
                  responseStatus: response.status
              });
              return response.json();
          })
          .then(response => {
              console.log(response);
              if(this.state.responseStatus === 200){
                  localStorage.setItem('TOKEN', response.accessToken);
                  this.props.onChangeFlag(true);
                  customHistory.push('/dashboard');
              } else {
                  console.log('ERROR: ', response);
              }
          })
          .catch(error => console.log('ERROR: ', error.message));
    };

Comment (obabichev) [reply src/components/SignIn/SignIn.jsx]: Just commit and push, PR will be updated

`if(this.state.responseStatus === 200)` - why do you take status from... `this`... why not from `response`?

**Pull request closed without merging**